### DISCUSSION TITLE
[FR] Prebuild X amount of pages and still use SSR for the rest

### DISCUSSION THREAD

**Discussion by goellner** [withastro/roadmap#641] [Proposal]
### Body

# Summary

Make it possible to combine getStaticPaths with SSR to allow prebuilding pages on build time and use SSR for the rest of the pages. Currently it is not possible to handle a large amount of pages without relying on SSR for all of them.

# Background & Motivation

Currently building a site, where we have 100k entries and where we want to prebuild lets say 1000 pages on build time. The rest of the pages should be generated with SSR on demand. When we use `hybrid` or `server` mode, Astro returns a 404, because it is not returned by `getStaticPaths`

# Goals

- make it possible to prerender pages on build time
- don't return 404 when a path is not in `getStaticPaths`, fall back to SSR

# Example

I have no example, since I don't know where to start

**Comment by lilnasy** [comment]
This is possible with an integration.

The API might change with 3.0, but currently you can use the `astro:build:setup` hook.

**Comment by goellner** [comment]
With this we can only prebuild the 1000 pages, but they wouldn't regenerate if data from the backend changes right? So the 1000 pages would remain static.

**Comment by lilnasy** [comment]
> but they wouldn't regenerate

That is correct, and it is an important feature. Your prerendered pages can use node built-in modules and libraries that your deployment runtime does not support.

If you want to update content without building from source while keeping the performance benefits of static builds, you are probably looking for ISR.

[Incremental Static Regeneration 🚀 Astro Documentation](https://deploy-preview-3527--astro-docs-2.netlify.app/en/guides/incremental-static-regeneration/)
**Note**: this guide is a work-in-progress that I am writing, and not part of our official docs.
